src/lib/builtins/array.rs

```diff
@@ -198,6 +198,24 @@ pub fn join(this: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
     Ok(to_value(elem_strs.join(&separator)))
 }
 
+/// Array.prototype.toString ( separator )
+///
+/// The toString function is intentionally generic; it does not require that
+/// its this value be an Array object. Therefore it can be transferred to
+/// other kinds of objects for use as a method.
+/// <https://tc39.es/ecma262/#sec-array.prototype.tostring>
+pub fn to_string(this: &Value, _args: &[Value], _ctx: &mut Interpreter) -> ResultValue {
+    let join_result = join(this, &[to_value(",")], _ctx);
+    let match_string = match join_result {
+        Ok(v) => match *v {
+            ValueData::String(ref s) => (*s).clone(),
+            _ => "".to_string(),
+        },
+        Err(v) => format!("{}: {}", "error", v.to_string()),
```

Comment:
No need for `"error"` to be a separate param, you can change the template to `"error: {}"`.

Also, no need for `.to_string()`, `v` will be formatted accordingly to its value anyway.